cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -17,53 +15,229 @@
+                bom = readJSONfile();
+                bom = addExternalReference(bom, url, comment);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
 
-        bom.addComponent(comp1);
+            case "addComponentExtRef": {                                 //This adds external Reference to Component
+                bom = readJSONfile();
+                bom = addComponentExternalReference(bom, url, hashes, comment);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
+            default: {
+                System.out.println("Please enter a command.");
+            }
+        }
+    }
 
+    static Bom createBom(final String name, final String version) {        //Create SBOM, test.JSON file
+        bom = new Bom();
+        System.out.println(bom.getBomFormat());
+        comp.setName(name);
+        comp.setVersion(version);
+        comp.setType(Component.Type.APPLICATION);
+        comp.setGroup("Eclipse Temurin");
+        comp.setAuthor("Vendor: Adoptium");
+        bom.addComponent(comp);
+        return bom;
+    }
+    static Bom addMetadata(final Bom bom, final String name) {               //Method to store metadata -->  name
+        comp.setName(name);
+        comp.setType(Component.Type.APPLICATION);
+        OrganizationalEntity org = new OrganizationalEntity();
+        org.setName("Eclipse Foundation");
+        org.setUrls(Collections.singletonList("https://www.eclipse.org/"));
+        meta.setManufacture(org);
+        meta.setComponent(comp);
+        bom.setMetadata(meta);
+        return bom;
+    }
+    static Bom addMetadataProperty(final Bom bom, final String name, final String value) {     //Method to store metadata --> Properties List --> name-values
+        meta = bom.getMetadata();
+        prop1.setName(name);
+        prop1.setValue(value);
+        meta.addProperty(prop1);
+        bom.setMetadata(meta);
+        return bom;
+    }
+
+    static Bom addComponent (final Bom bom, final String name) {                    //Method to store Component -> name
+        comp.setType(Component.Type.APPLICATION);
+        comp.setName(name);
+        bom.addComponent(comp);
+        return bom;
+    }
+    static Bom addComponentProperty(final Bom bom, final String name, final String value) {     //Method to store Component-> Property-> name-value pairs
+        prop1.setName(name);
+        prop1.setValue(value);
+        prop.add(prop1);
+        comp.setProperties(prop);
+        bom.addProperty(prop1);
+        return bom;
+    }
+    static Bom addExternalReference(final Bom bom, final String url, final String comment) {   //Method to store externalReferences: dependency_version_alsa
+        hash.add(hash1);
```

Comment:
and here:
```
ExternalReference extRef = new ExternalReference();
```
Remove the static global variables.
The "art" of good class structure is "encapsulation", eg.creating variables within scope for the purpose of just that method.